optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/spotroster/SpotRosterPage.java

```diff
@@ -186,6 +191,10 @@ public void onTenantChanged(@Observes final TenantStore.TenantChange tenant) {
         return refreshWithoutLoadingSpinner().then(i -> {
             loadingSpinner.hideFor("refresh-spot-roster");
             return resolve();
+        }).catch_(e -> {
+            ErrorPopup.show(e.toString());
```

Comment:
Now this is a piece of code that can use a comment :)
No code should every call ErrorPopup.show() directly, because that's being handled as "an aspect" (which is a fancy way of saying that there's global exception handler that makes sure the pop up is shown).
But I bet there's a good reason to do it manually here  (the loadingSpinner.hide maybe?). Add a comment to explain that good reason.

Reply:
It appears the global exception handler does not work with promises, so to actually show the error, it need to be explicitly catched. Plus, we need to hide the loading spinner; I add that as a comment.